src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -501,23 +503,33 @@ bool layout::identical(const layout& other) const {
     return ret;
 }
 
-ov::PartialShape layout::transform(const ov::PartialShape& pshape, cldnn::format old_fmt, cldnn::format new_fmt) {
+ov::PartialShape layout::transform(const ov::PartialShape& pshape, const cldnn::format& old_fmt, const cldnn::format& new_fmt) {
     if (old_fmt == new_fmt) {
         return pshape;
     }
 
+    // shortcut for transform to max rank default fmt which is used in fill_shape_info_data to improve perf
+    if (format::is_default_format(old_fmt) && new_fmt == format::bfvuwzyx) {
+        ov::PartialShape res = pshape;
+        size_t num_to_insert = layout::max_rank() - pshape.size();
+        size_t pos_to_insert = pshape.size() > 1 ? 2 : 1;
```

Comment:
Are there any guarantees that we will not have scalar and empty pshape here? Maybe it makes sense to add pshape.size() check as well?

Reply:
updated this part to min(pshape.size(), 2)